python/ray/_private/runtime_env/conda.py

```diff
@@ -265,34 +290,41 @@ def delete_uri(self,
         return successful
 
     def setup(self,
-              runtime_env: Dict,
+              runtime_env: RuntimeEnv,
+              serialized_runtime_env: str,
               context: RuntimeEnvContext,
+              uri: str,
```

Comment:
Emm let me see. The reason of adding ｀serialized_runtime_env` is to avoid serializing again. And the reason of adding ｀uri｀is to walk around a circle dependency issue. Let me try to clean this.

Reply:
I see--actually, thinking about it again, I think you're right to put the URI here, but either way should be fine for this PR.  Actually we'll soon split `setup` into two functions `create` and `modify_context` to match this doc https://docs.google.com/document/d/18X8rx6Q-vYj6FyYWQEjqqnH0KRx97-OrTDuw7NwV5PM/edit#, and they'll take in URI as an argument so that the URI caching logic can be handled in the agent (outside of the plugin implementation.)

Reply:
OK, make sense. By the way, I don't have the permission of your doc. Please add me and @raulchen.

Reply:
Ah sorry, wrong link, I updated the link--let me know if it still doesn't work